rclcpp/test/rclcpp/executors/executor_types.hpp

```diff
@@ -32,6 +32,12 @@ using ExecutorTypes =
   rclcpp::executors::StaticSingleThreadedExecutor,
   rclcpp::experimental::executors::EventsExecutor>;
 
+using MainExecutorTypes =
+  ::testing::Types<
+  rclcpp::executors::SingleThreadedExecutor,
+  rclcpp::executors::MultiThreadedExecutor,
+  rclcpp::executors::StaticSingleThreadedExecutor>;
```

Comment:
Given that this is only used in `test_executors_timer_cancel_behavior.cpp` for now, I think we can just move it there.  If it turns out we need it in other tests, we can always move it later.